components/brave_wallet_ui/components/buy-send-swap/buy/index.tsx

```diff
@@ -36,20 +49,52 @@ function Buy (props: Props) {
     defaultCurrencies,
     onInputChange,
     onSubmit,
-    onChangeBuyView
+    onChangeBuyView,
+    selectedBuyOption,
+    onSelectBuyOption,
+    wyreAssetOptions,
+    rampAssetOptions
   } = props
+  const [buyOptions, setBuyOptions] = React.useState<BuyOption[]>(BuyOptions)
 
   const onShowAssets = () => {
     onChangeBuyView('assets', 'from')
   }
 
+  React.useEffect(() => {
+    const supportingBuyOptions = BuyOptions.filter(buyOption => {
+      if (buyOption.id === BraveWallet.OnRampProvider.kWyre) {
+        return isSelectedAssetInAssetOptions(selectedAsset, wyreAssetOptions)
+      }
+
+      if (buyOption.id === BraveWallet.OnRampProvider.kRamp) {
+        return isSelectedAssetInAssetOptions(selectedAsset, rampAssetOptions)
+      }
+
+      return false
+    })
+    setBuyOptions(supportingBuyOptions)
+  }, [selectedAsset, wyreAssetOptions, rampAssetOptions])
+
+  React.useEffect(() => {
+    if (buyOptions.length > 0) {
+      onSelectBuyOption(buyOptions[0]?.id)
+    }
+  }, [buyOptions])
+
   const networkName = React.useMemo((): string => {
     return networkList.find((network) => network.chainId === selectedNetwork.chainId)?.chainName ?? ''
   }, [networkList, selectedNetwork])
 
+  const buyWithLabel = React.useMemo(() => {
+    const selected = buyOptions.find(option => option.id === selectedBuyOption)
+
+    return selected !== undefined ? selected.label : ''
+  }, [selectedBuyOption])
+
   return (
     <StyledWrapper>
-      {selectedNetwork.chainId === BraveWallet.MAINNET_CHAIN_ID ? (
+      {BuySupportedChains.includes(selectedNetwork.chainId) ? (
```

Comment:
Not very important, but can we simply check if `selectedNetwork.chainId` is in at least one of ramp or wyre assets?